E-periodica Switzerland.js

```diff
@@ -0,0 +1,294 @@
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes('/digbib/view')) {
+		return "journalArticle";
+	}
+	else if (url.includes('/digbib/doasearch') && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	else {
+		return false;
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('h2.ep-result__title > a');
+	for (let row of rows) {
+		// Zotero.debug(row.innerHTML);
+		let href = row.href;
+		Zotero.debug(href);
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'journalArticle') {
+		await scrape(doc, url);
+	}
+	else if (detectWeb(doc, url) == 'multiple') {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		// The fallback is not expected to be used on E-periodica, but just in case...
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(nextDoc, url) {
+	var nextUrl = nextDoc.location.href;
+	Zotero.debug('trying to process ' + nextUrl);
+	// Do we really need to handle these #-containing URLs?
+	nextUrl = nextUrl.replace("#", "%3A%3A").replace("::", "%3A%3A");
+	nextUrl = nextUrl.split("%3A%3A");
+	nextUrl = nextUrl[0] + "%3A%3A" + nextUrl.slice(-1);
+	Zotero.debug('Final URL ' + nextUrl);
+	var pageinfoUrl = nextUrl.replace("view", "ajax/pageinfo");
+	Zotero.debug('JSON URL ' + pageinfoUrl);
+	let text = await requestText(pageinfoUrl);
+	var epJSON = JSON.parse(text);
+	// Zotero.debug(epJSON);
+	var risURL = null;
+	if (epJSON.articles["0"].hasRisLink) {
+		risURL = '/view/' + epJSON.articles["0"].risLink;
+	}
+	
+	Zotero.debug(risURL);
+	var pdfURL = null;
+	if (epJSON.articles["0"].hasPdfLink) {
+		pdfURL = epJSON.articles["0"].pdfLink;
+	}
+	
+	Zotero.debug(pdfURL);
+	if (risURL) {
+		let text = await requestText(risURL);
+		processRIS(text, url, pdfURL);
+	}
+	else {
+		var item = new Zotero.Item("journalArticle");
+		item.title = epJSON.articles["0"].title.replace(' : ', ': ');
+		item.publicationTitle = epJSON.journalTitle.replace(' : ', ': ');
+		var numyear = epJSON.volumeNumYear.replace("(", "").replace(")", "").split(" ");
+		if (numyear.length > 1) {
+			item.date = numyear.slice(-1);
+		}
+		if (numyear.length > 0) {
+			item.volume = numyear[0];
+		}
+		if (pdfURL) {
+			Zotero.debug('PDF URL: ' + pdfURL);
+			item.attachments.push({
+				url: pdfURL,
+				title: "E-periodica PDF",
```

Comment:
```suggestion
				title: "Full Text PDF",
```

We use this consistently across translators